API reference: new settings page (#1795)
meilisearch/documentation: reference/api/settings.md

---
sidebarDepth: 2
---

# Settings

The `/settings` route allows you to customize search settings for the given index. You can either modify all of an index's settings at once using the [update settings endpoint](#update-settings), or modify each one individually using the child routes.

For a conceptual overview of index settings, refer to our [indexes guide](/learn/core_concepts/indexes.md#index-settings).

## Settings object

By default, the settings object looks like this. All fields are modifiable.

```json
{
  "displayedAttributes":[
    "*"
  ],
  "searchableAttributes":[
    "*"
  ],
  "filterableAttributes":[],
  "sortableAttributes":[],
  "rankingRules":
  [
    "words",
    "typo",
    "proximity",
    "attribute",
    "sort",
    "exactness"
  ],
  "stopWords":[],
  "synonyms":{},
  "distinctAttribute":null,
  "typoTolerance":{
    "enabled":true,
    "minWordSizeForTypos":{
      "oneTypo":5,
      "twoTypos":9
    },
    "disableOnWords":[],
    "disableOnAttributes":[]
    },
  "faceting":{
      "maxValuesPerFacet":100
      },
  "pagination":{
      "maxTotalHits":1000
      }
}
```

> To learn more about any one of these fields, click on its name in the sidebar.

Reviewer: I find it a bit clunky when a text directly refers to a UI element that's outside the regular content flow. It might also be a bit not future-proof if we change the content on the sidebar menu somewhere else on the screen on the new design. 🤔

  "rankingRules":
  [
    "words",
    "typo",
    "proximity",
    "attribute",
    "sort",
    "exactness"
  ],
  "stopWords": [],
  "synonyms": {},
  "distinctAttribute": null,
  "typoTolerance": {
    "enabled": true,
    "minWordSizeForTypos": {
      "oneTypo": 5,
      "twoTypos": 9
    },
    "disableOnWords": [],
    "disableOnAttributes": []
  },
  "faceting": {
    "maxValuesPerFacet": 100
  },
  "pagination": {
    "maxTotalHits": 1000
  }
}
```

### Update settings

<RouteHighlighter method="PATCH" route="/indexes/{index_uid}/settings" />

Update the settings of an index.

Passing `null` to an index setting will reset it to its default value.

Updates in the settings route are **partial**. This means that any parameters not provided in the body will be left unchanged.

If the provided index does not exist, it will be created.

#### Path parameters

| Name        | Type   | Description                                                               |
| :---------- | :----- | :------------------------------------------------------------------------ |
| **`uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |

> #### Body

Author: This is because of the default value column. I could add `</br>` for ranking rules and typo tolerance for now

### Reset settings

<RouteHighlighter method="DELETE" route="/indexes/{index_uid}/settings"/>

Reset all the settings of an index to their [default value](#settings-object).

#### Path parameters

| Name              | Type   | Description                                                               |
| :---------------- | :----- | :------------------------------------------------------------------------ |
| **`index_uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |

#### Example

<CodeSamples id="reset_settings_1" />

##### Response: `202 Accepted`

```json
{
  "taskUid": 1,
  "indexUid": "movies",
  "status": "enqueued",
  "type": "settingsUpdate",
  "enqueuedAt": "2021-08-11T09:25:53.000000Z"
}
```

You can use this `taskUid` to get more details on [the status of the task](/reference/api/tasks.md#get-one-task).

## Displayed attributes

The attributes added to the `displayedAttributes` list are displayed in search results. `displayedAttributes` only affects the search endpoints; it has no impact on the [GET documents endpoint](/reference/api/documents.md#get-documents).

> By default, all fields are considered to be `displayedAttributes`. This behavior is represented by the `*` in the settings. Setting `displayedAttributes` to an empty array `[]` will reset the setting to its default value.

Reviewer 2: ```suggestion
By default, all fields are considered to be `displayedAttributes`. This behavior is represented by the `["*"]` in the settings.
```

##### Response: `202 Accepted`

```json
{
  "taskUid": 1,
  "indexUid": "books",
  "status": "enqueued",
  "type": "settingsUpdate",
  "enqueuedAt": "2022-04-14T20:56:44.991039Z"
}
```

You can use the returned `taskUid` to get more details on [the status of the task](/reference/api/tasks.md#get-one-task).

### Reset pagination settings

Reset an index's pagination settings to their [default value](#pagination-object).

#### Path parameters

| Name              | Type   | Description                                                               |
| :---------------- | :----- | :------------------------------------------------------------------------ |
| **`index_uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |

#### Example

<CodeSamples id="reset_pagination_settings_1" />

##### Response: `202 Accepted`

```json
{
  "taskUid": 1,
  "indexUid": "books",
  "status": "enqueued",
  "type": "settingsUpdate",
  "enqueuedAt": "2022-04-14T20:53:32.863107Z"
}
```

You can use the returned `taskUid` to get more details on [the status of the task](/reference/api/tasks.md#get-one-task).

## Ranking rules

> Ranking rules are built-in rules that allow you to customize the relevancy of your search results. They are stored in an array and applied in order of appearance.

Reviewer 2: I found "order of appearance" potentially confusing so this is my attempt to improve it. Feel free to veto if you disagree

```suggestion
Ranking rules are built-in rules that rank search results according to certain criteria. They are applied in the same order in which they appear in the `rankingRules` array.
```

To learn more about ranking rules, refer to our [dedicated guide](/learn/core_concepts/relevancy.md).

### Ranking rules array

| Name              | Description                                                                     |
| :---------------- | :------------------------------------------------------------------------------ |
| **`"words"`**     | Sorts results by decreasing number of matched query terms                       |
| **`"typo"`**      | Sorts results by increasing number of typos                                     |
| **`"proximity"`** | Sorts results by increasing distance between matched query terms                |
| **`"attribute"`** | Sorts results based on the attribute ranking order                              |
| **`"sort"`**      | Sorts results based on parameters decided at query time                         |

> | **`"exactness"`** | Sorts results based on the similarity of the matched words with the query words |

Reviewer 2: Absolutely love this, it's such a huge improvement to have this here ❤️

{
  "taskUid": 1,
  "indexUid": "movies",
  "status": "enqueued",
  "type": "settingsUpdate",
  "enqueuedAt": "2021-08-11T09:25:53.000000Z"
}
```

You can use this `taskUid` to get more details on [the status of the task](/reference/api/tasks.md#get-one-task).

### Reset searchable attributes

<RouteHighlighter method="DELETE" route="/indexes/{index_uid}/settings/searchable-attributes"/>

Reset the searchable attributes of the index to the default value.

#### Path parameters

| Name              | Type   | Description                                                               |
| :---------------- | :----- | :------------------------------------------------------------------------ |
| **`index_uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |

#### Example

<CodeSamples id="reset_searchable_attributes_1" />

##### Response: `202 Accepted`

```json
{
  "taskUid": 1,
  "indexUid": "movies",
  "status": "enqueued",
  "type": "settingsUpdate",
  "enqueuedAt": "2021-08-11T09:25:53.000000Z"
}
```

You can use this `taskUid` to get more details on [the status of the task](/reference/api/tasks.md#get-one-task).

## Sortable attributes

> Attributes used when sorting search results. `sortableAttributes` can be used together with the [`sort` search parameter](/reference/api/search.md#sort).

Reviewer 2: ```suggestion
Attributes that can be used when sorting search results using the [`sort` search parameter](/reference/api/search.md#sort).
```

To learn more about sortable attributes, refer to our [dedicated guide](/learn/advanced/sorting.md).

### Get sortable attributes

<RouteHighlighter method="GET" route="/indexes/{index_uid}/settings/sortable-attributes" />

Get the sortable attributes of an index.

#### Path parameters

| Name              | Type   | Description                                                               |
| :---------------- | :----- | :------------------------------------------------------------------------ |
| **`index_uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |

#### Example

<CodeSamples id="get_sortable_attributes_1" />

##### Response: `200 Ok`

List the settings.

```json
[
  "price", 
  "author.surname"
]
```

### Update sortable attributes

<RouteHighlighter method="PUT" route="/indexes/{index_uid}/settings/sortable-attributes" />

Update an index's sortable attributes list.

If an attribute contains an object, you can use dot notation to set one or more of its keys as a value for this setting: `"sortableAttributes": ["author.surname"]`.

::: warning
If the field does not exist, no error will be thrown.

> :::

Reviewer 2: Let's move these to the body section.

[You can read more about sorting at query time on our dedicated guide.](/learn/advanced/sorting.md)

#### Path parameters

| Name              | Type   | Description                                                               |
| :---------------- | :----- | :------------------------------------------------------------------------ |
| **`index_uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |

#### Body

> An array of strings containing the attributes that can be used to sort search results at query time.

Reviewer 2: ```suggestion
An array of strings. Each string should be an attribute that exists in the selected index.
```

```json
{
  "taskUid": 1,
  "indexUid": "movies",
  "status": "enqueued",
  "type": "settingsUpdate",
  "enqueuedAt": "2021-08-11T09:25:53.000000Z"
}
```

You can use this `taskUid` to get more details on [the status of the task](/reference/api/tasks.md#get-one-task).

### Reset sortable attributes

<RouteHighlighter method="DELETE" route="/indexes/{index_uid}/settings/sortable-attributes"/>

Reset an index's sortable attributes list back to its default value.

#### Path parameters

| Name              | Type   | Description                                                               |
| :---------------- | :----- | :------------------------------------------------------------------------ |
| **`index_uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |

#### Example

<CodeSamples id="reset_sortable_attributes_1" />

##### Response: `202 Accepted`

```json
{
  "taskUid": 1,
  "indexUid": "movies",
  "status": "enqueued",
  "type": "settingsUpdate",
  "enqueuedAt": "2021-08-11T09:25:53.000000Z"
}
```

You can use this `taskUid` to get more details on [the status of the task](/reference/api/tasks.md#get-one-task).

## Stop words

> Adding words to the stop words list causes them to be ignored in future search queries. Small words that appear in almost all of a dataset's documents are prime candidates for this list.

Reviewer 2: ```suggestion
Words added to the `stopWords` list are ignored in future search queries.
```

:::

### Get stop words

<RouteHighlighter method="GET" route="/indexes/{index_uid}/settings/stop-words" />

Get the stop words list of an index.

#### Path parameters

| Name              | Type   | Description                                                               |
| :---------------- | :----- | :------------------------------------------------------------------------ |
| **`index_uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |

#### Example

<CodeSamples id="get_stop_words_1" />

##### Response: `200 Ok`

```json
[
  "of",
  "the",
  "to"
]
```

### Update stop words

<RouteHighlighter method="PUT" route="/indexes/{index_uid}/settings/stop-words" />

Update the list of stop words of an index.

#### Path parameters

| Name              | Type   | Description                                                               |
| :---------------- | :----- | :------------------------------------------------------------------------ |
| **`index_uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |

#### Body

> An array of strings that contains the stop words.

Reviewer 2: ```suggestion
An array of strings. Each string should be a single word.
```

Get the list of synonyms of an index.

#### Path parameters

| Name              | Type   | Description                                                               |
| :---------------- | :----- | :------------------------------------------------------------------------ |
| **`index_uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |

#### Example

<CodeSamples id="get_synonyms_1" />

##### Response: `200 OK`

```json
{
  "wolverine": [
    "xmen",
    "logan"
  ],
  "logan": [
    "wolverine",
    "xmen"
  ],
  "wow": [
    "world of warcraft"
  ]
}
```

### Update synonyms

<RouteHighlighter method="PUT" route="/indexes/{index_uid}/settings/synonyms"/>

Update the list of synonyms of an index. Synonyms are [normalized](/learn/configuration/synonyms.md#normalization).

#### Path parameters

| Name              | Type   | Description                                                               |
| :---------------- | :----- | :------------------------------------------------------------------------ |
| **`index_uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |

#### Body

> An object that contains all synonyms and their associated words.

Reviewer 2: If possible, I think it would be helpful to add a brief example/explanation of the form that the synonym object takes, so that the user can understand how to structure the object without clicking the below link. On the other hand, there is already a helpful example below so this may be unnecessary labor.

I'm genuinely not sure if it's necessary so I leave the decision to you Maryam, as the person who would actually have to write it 😄

---
sidebarDepth: 2
---

# Settings

The `/settings` route allows you to customize search settings for the given index. You can either modify all of an index's settings at once using the [update settings endpoint](#update-settings), or modify each one individually using the child routes.

::: warning
When you update a setting, you overwrite its default value. Use the `DELETE` route to reset any setting to its original value.
:::

## Settings object

The default settings object:

> ```json

Author: I used an [online tool](https://jsonformatter.curiousconcept.com/#) but forgot to specify the indentation

{
    "displayedAttributes":[
        "*"
    ],
    "searchableAttributes":[
        "*"
    ],
    "filterableAttributes":[],
    "sortableAttributes":[],
    "rankingRules":
    [
        "words",
        "typo",
        "proximity",
        "attribute",
        "sort",
        "exactness"
    ],
    "stopWords":[],
    "synonyms":{},
    "distinctAttribute":null,
    "typoTolerance":{
        "enabled":true,
        "minWordSizeForTypos":{
            "oneTypo":5,
            "twoTypos":9
            },
        "disableOnWords":[],
        "disableOnAttributes":[]
        },
    "faceting":{
        "maxValuesPerFacet":100
        },
    "pagination":{
        "maxTotalHits":1000
        }
}
```

## All settings

This route lets you configure all the settings at once for an index.

### Get settings

<RouteHighlighter method="GET" route="/indexes/{index_uid}/settings" />

> Get the settings of an index. The index [`uid`](/learn/core_concepts/indexes.md#index-uid) is required.

Reviewer 2: Do you think the phrase `The index [uid](..) is required.` is adding anything of value? I feel like it's not and it's repeated 36 times on this page. I'd like to get rid of it. If we wanted, we could add a short paragraph or note box to this effect at the top of the page, since it applies equally to all settings endpoints.

I appreciate the goal of specifying path variables for all endpoints, but from a practical perspective I don't think this is worth the space. The variable is called `index_uid` so I don't see it causing any confusion as long as the user can see the `routeHighlighter` element—and if they can't, we have bigger problems.

Author: I looked at other API documentation, and they document path parameters in a lot more detail. But we can get rid of them if we go with [highlighting them](https://github.com/meilisearch/documentation/issues/1808)

Reviewer 2: Indeed, [it is the norm to document path parameters](https://idratherbewriting.com/learnapidoc/docapis_doc_parameters.html#path_parameters), usually in a table. But I remember what we discussed when we removed the tables: the only description for `index_uid` and `task_uid` that we came up with was something like "The index/task UID", which is obviously unhelpful. Also, very few of our endpoints have more than one path parameter, so it was mostly just a bunch of tables with a single row, taking up space without adding information.

Three options:
- We follow the norm and re-implement path parameter tables
- We keep things how they are: a single sentence in the endpoint description
- We try removing the aforementioned sentence and trust our users to find that information themselves (e.g. if you're wondering what `index_uid` is, you click on `indexes` in the sidebar and are greeted with the index object definition—pretty intuitive).

The first option is the most extensible, since it adapts well if we ever have endpoints with many path parameters, or path parameters that are not self-explanatory.

The last option is the most efficient, as it saves space and makes the endpoint descriptions easier to parse.

The middle option (currently implemented) provides no particular benefits IMO. It is not very extensible, and it makes the endpoint descriptions harder to parse.

WDYT @guimachiavelli @maryamsulemani97 ?

Reviewer: @dichotommy, what do you think?

Author: We'll be using Path parameter tables now, as discussed yesterday

#### Example

<CodeSamples id="get_settings_1" />

#### Response: `200 Ok`

> ```json

Reviewer 2: I notice that these are not at all in the same order as the settings object above. This response is probably outdated. Could you fix it, please?

| **[`sortableAttributes`](#sortable-attributes)**     | Array of strings | Empty                                                                                                                                                    | Attributes to use when sorting search results                                    |
| **[`stopWords`](#stop-words)**                       | Array of strings | Empty                                                                                                                                                    | List of words ignored by Meilisearch when present in search queries              |
| **[`synonyms`](#synonyms)**                          | Object           | Empty                                                                                                                                                    | List of associated words treated similarly                                       |

> | **[`typoTolerance`](#typo-tolerance)**               | Object           | `{"enabled": true,`</br>`"minWordSizeForTypos":`</br>`{"oneTypo": 5,`</br>`"twoTypos": 9},`</br>`"disableOnWords": [],`</br>`"disableOnAttributes": []}` | Typo tolerance settings                                                          |

Author: Agree, but I think the line breaks help 🫣

![Screen Shot 2022-08-18 at 12 48 03](https://user-images.githubusercontent.com/90181761/185352827-8d2faa3e-5f08-4cb6-b5d3-eff7f47db1c6.png)

You can use this `taskUid` to get more details on [the status of the task](/reference/api/tasks.md#get-one-task).

### Reset settings

<RouteHighlighter method="DELETE" route="/indexes/{index_uid}/settings"/>

Reset all the settings of an index to their [default value](#settings-object).

#### Path parameters

| Name        | Type   | Description                                                               |
| :---------- | :----- | :------------------------------------------------------------------------ |
| **`uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |

#### Example

<CodeSamples id="reset_settings_1" />

##### Response: `202 Accepted`

```json
{
    "taskUid": 1,
    "indexUid": "movies",
    "status": "enqueued",
    "type": "settingsUpdate",
    "enqueuedAt": "2021-08-11T09:25:53.000000Z"
}
```

You can use this `taskUid` to get more details on [the status of the task](/reference/api/tasks.md#get-one-task).

## Displayed attributes

> The fields whose attributes are added to the displayed attributes list are displayed in each matching document.

Reviewer 2: I think it would be good to move the note that is currently in the update displayed attributes endpoint up here. Also making a suggestion to simplify the language.

```suggestion
The attributes added to the `displayedAttributes` list are displayed in search results. `displayedAttributes` only affects the search endpoints; it has no impact on the [GET documents endpoint](/reference/api/documents.md#get-documents).
```

> By default, all fields are considered to be `displayedAttributes`. This behavior is represented by the `*` in the settings. Setting `displayedAttributes` to an empty array `[]` will reset the setting to its default value.

Reviewer 2: I think this paragraph should be moved or removed. Consider similar settings like `searchableAttributes`, which also has default value `["*"]`. Where do we mention this information for that setting? If we don't, where do you think would be the best place to mention this for both settings?

```suggestion
```

Author: I think it would be better to add this for `searchableAttributes` too. Users will probably be looking at each setting separately

To learn more about displayed attributes, refer to our [dedicated guide](/learn/configuration/displayed_searchable_attributes.md#displayed-fields).

### Get displayed attributes

<RouteHighlighter method="GET" route="/indexes/{index_uid}/settings/displayed-attributes" />

Get the displayed attributes of an index.

#### Path parameters

| Name        | Type   | Description                                                               |
| :---------- | :----- | :------------------------------------------------------------------------ |
| **`uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |

#### Example

<CodeSamples id="get_displayed_attributes_1"/>

##### Response: `200 Ok`

```json
[
  "title",
  "overview",
  "genres",
  "release_date.year"
]
```

### Update displayed attributes

<RouteHighlighter method="PUT" route="/indexes/{index_uid}/settings/displayed-attributes" />

Update the displayed attributes of an index.

If an attribute contains an object, you can use dot notation to specify one or more of its keys, e.g., `"displayedAttributes": ["release_date.year"]`.

> 
> ::: note
> `displayedAttributes` only impacts search results. It has no effect on other methods of retrieving or copying documents, such as the [GET documents endpoint](/reference/api/documents.md#get-documents), [dumps](/learn/advanced/dumps.md), or [snapshots](/learn/advanced/snapshots.md).
> :::

Reviewer 2: If you accept my above suggestion:

```suggestion
```

```json
{
    "taskUid": 1,
    "indexUid": "movies",
    "status": "enqueued",
    "type": "settingsUpdate",
    "enqueuedAt": "2021-08-11T09:25:53.000000Z"
}
```

You can use this `taskUid` to get more details on [the status of the task](/reference/api/tasks.md#get-one-task).

### Reset sortable attributes

<RouteHighlighter method="DELETE" route="/indexes/{index_uid}/settings/sortable-attributes"/>

Reset an index's sortable attributes list back to its default value.

#### Path parameters

| Name        | Type   | Description                                                               |
| :---------- | :----- | :------------------------------------------------------------------------ |
| **`uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |

#### Example

<CodeSamples id="reset_sortable_attributes_1" />

##### Response: `202 Accepted`

```json
{
    "taskUid": 1,
    "indexUid": "movies",
    "status": "enqueued",
    "type": "settingsUpdate",
    "enqueuedAt": "2021-08-11T09:25:53.000000Z"
}
```

You can use this `taskUid` to get more details on [the status of the task](/reference/api/tasks.md#get-one-task).

## Stop words

> This route allows you to add a list of words ignored in your search queries. During a search, the stop words contained in your search query will be ignored by the sorting algorithm. When you add a common English word such as `the` to the stop words list, Meilisearch will not take it into consideration when calculating how relevant a result is.

Reviewer 2: I would like to shorten this paragraph to 1–2 brief, efficient sentences. This is just a suggestion, so feel free to rewrite it!

```suggestion
Adding words to the stop words list causes them to be ignored in future search queries. Small words that appear in almost all of a dataset's documents are prime candidates for this list.
```

##### Response: `202 Accepted`

```json
{
    "taskUid": 1,
    "indexUid": "movies",
    "status": "enqueued",
    "type": "settingsUpdate",
    "enqueuedAt": "2021-08-11T09:25:53.000000Z"
}
```

You can use this `taskUid` to get more details on [the status of the task](/reference/api/tasks.md#get-one-task).

### Reset stop words

<RouteHighlighter method="DELETE" route="/indexes/{index_uid}/settings/stop-words" />

Reset the list of stop words of an index to its default value.

#### Path parameters

| Name        | Type   | Description                                                               |
| :---------- | :----- | :------------------------------------------------------------------------ |
| **`uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |

#### Example

<CodeSamples id="reset_stop_words_1" />

##### Response: `202 Accepted`

```json
{
    "taskUid": 1,
    "indexUid": "movies",
    "status": "enqueued",
    "type": "settingsUpdate",
    "enqueuedAt": "2021-08-11T09:25:53.000000Z"
}
```

You can use this `taskUid` to get more details on [the status of the task](/reference/api/tasks.md#get-one-task).

## Synonyms

> The `synonyms` object contains words and their respective synonyms. A synonym in Meilisearch is considered equal to its associated word in a search query. This route allows you to manage synonyms for an index.

Reviewer 2: I would like to shorten this paragraph to 1–2 brief, efficient sentences. This is just a suggestion, so feel free to rewrite it!

```suggestion
The `synonyms` object contains words and their respective synonyms. A synonym in Meilisearch is considered equal to its associated word for the purposes of calculating search results.
```

The values associated with attributes in the `searchableAttributes` list are searched for matching query words. The order of the list also determines the [attribute ranking order](/learn/core_concepts/relevancy.md#attribute-ranking-order).

> By default, all fields are part of the `searchableAttributes` array. This behavior is represented by the `*` in the settings. Setting `searchableAttributes` to an empty array `[]` will reset the setting to its default value.

Author: You suggested removing it for [displayed attributes](https://github.com/meilisearch/documentation/pull/1795/files#r947123457). I couldn't think of where we could mention this for both settings, so I added it for `searchableAttributes` too, as users will probably be looking at each setting individually. 
We could mention this on the API/Overview page when we get to it.

> By default, all fields are part of the `searchableAttributes` array. This behavior is represented by the `*` in the settings. Setting `searchableAttributes` to an empty array `[]` will reset the setting to its default value.

Reviewer 2: I see! Okay, I respect your decision. I would recommend though that we remove the sentence about setting it to an empty array as a way of resetting. To me, this is one of those cases where presenting too many equivalent options to the user creates needless confusion (sort of like having both GET and POST search endpoints is confusing—see screenshot). We already offer endpoints _specifically_ for resetting these settings to their default values, so I don't think it's necessary to complicate that.

However, if we want to be complete w/ our documentation and give users the option to reset using the update endpoint, we could simply move this sentence to the body section for both endpoints. Who knows, if enough users reset using the update endpoint, we may eventually be able to remove the reset endpoints 🤷🏻

> By default, all fields are part of the `searchableAttributes` array. This behavior is represented by the `*` in the settings. Setting `searchableAttributes` to an empty array `[]` will reset the setting to its default value.

Reviewer 2: ```suggestion
By default, all fields are part of the `searchableAttributes` array. This behavior is represented by the `["*"]` in the settings.
```
